### ISSUE TITLE
When my VictoriaLogs service does not add data and I call the query interface to use the statistics function, the service will panic.

### ISSUE THREAD

**Issue by zoutao0909** [VictoriaMetrics/VictoriaMetrics#6324]
### Describe the bug

When my VictoriaLogs service does not add data and I call the query interface to use the statistics function, the service will panic.

victoriaLogs version is 0.8, the request used is: curl http://localhost:9248/select/logsql/query -d 'query=* | stats count() logs'

The return value obtained is as follows:
curl: (52) Empty reply from server

View the victoriaLogs log as follows:
panic: runtime error: index out of range [0] with length 0

goroutine 58 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:291 +0x5b
panic({0x955280?, 0xc0001d8228?})
	runtime/panic.go:770 +0x132
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*pipeStatsProcessor).flush(0xc0000e65a0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/pipe_stats.go:431 +0xf2f
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).runQuery(0xc0000262c0, {0xb46400, 0xc0000e6460}, {0xc000014480, 0x1, 0x1}, 0xc0000c6600, 0xc00002e2c0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage_search.go:135 +0x50f
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).RunQuery(0xc0000262c0, {0xb46400, 0xc0000e6460}, {0xc000014480, 0x1, 0x1}, 0x7f6bf50a4108?, 0xc00002e280)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage_search.go:100 +0xba
github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage.RunQuery(...)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage/main.go:111
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql.ProcessQueryRequest({0xb46400, 0xc0000e6460}, {0xb44c58, 0xc0002164c8}, 0xc0001d0120)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql/logsql.go:239 +0x3a5
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect.RequestHandler({0xb44c58, 0xc0002164c8}, 0xc0001d0120)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/main.go:147 +0x99e
main.requestHandler({0xb44c58, 0xc0002164c8}, 0xc0001d0120)
	github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-logs/main.go:92 +0x195
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc00002e860, {0xb45708, 0xc000111380}, 0xc0001d0120, 0xac0928)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:416 +0x1003
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xb45708?, 0xc000111380?}, 0xc000111380?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:250 +0x31
net/http.HandlerFunc.ServeHTTP(0x961460?, {0xb45708?, 0xc000111380?}, 0x4?)
	net/http/server.go:2166 +0x29
github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xb44b38, 0xc00023e000}, 0xc0001d0120)
	github.com/klauspost/compress@v1.17.8/gzhttp/compress.go:501 +0x453
net/http.HandlerFunc.ServeHTTP(0x40eca5?, {0xb44b38?, 0xc00023e000?}, 0xc00023e001?)
	net/http/server.go:2166 +0x29
net/http.serverHandler.ServeHTTP({0xb439a0?}, {0xb44b38?, 0xc00023e000?}, 0x6?)
	net/http/server.go:3137 +0x8e
net/http.(*conn).serve(0xc00021c090, {0xb463c8, 0xc0000c6360})
	net/http/server.go:2039 +0x5e8
created by net/http.(*Server).Serve in goroutine 57
	net/http/server.go:3285 +0x4b4

### To Reproduce

1. Start the new victoriaLogs service
2. Execute the query request: curl http://localhost:9248/select/logsql/query -d 'query=* | stats count() logs'
3. Check the status and log of the victoriaLogs service and find that the service process has ended and there is a panic error in the log.

### Version

victoria-logs-20240520-021031-tags-v0.8.0-victorialogs-0-gad505a7a9a

### Logs

panic: runtime error: index out of range [0] with length 0

goroutine 58 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:291 +0x5b
panic({0x955280?, 0xc0001d8228?})
	runtime/panic.go:770 +0x132
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*pipeStatsProcessor).flush(0xc0000e65a0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/pipe_stats.go:431 +0xf2f
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).runQuery(0xc0000262c0, {0xb46400, 0xc0000e6460}, {0xc000014480, 0x1, 0x1}, 0xc0000c6600, 0xc00002e2c0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage_search.go:135 +0x50f
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).RunQuery(0xc0000262c0, {0xb46400, 0xc0000e6460}, {0xc000014480, 0x1, 0x1}, 0x7f6bf50a4108?, 0xc00002e280)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage_search.go:100 +0xba
github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage.RunQuery(...)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage/main.go:111
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql.ProcessQueryRequest({0xb46400, 0xc0000e6460}, {0xb44c58, 0xc0002164c8}, 0xc0001d0120)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql/logsql.go:239 +0x3a5
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect.RequestHandler({0xb44c58, 0xc0002164c8}, 0xc0001d0120)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/main.go:147 +0x99e
main.requestHandler({0xb44c58, 0xc0002164c8}, 0xc0001d0120)
	github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-logs/main.go:92 +0x195
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc00002e860, {0xb45708, 0xc000111380}, 0xc0001d0120, 0xac0928)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:416 +0x1003
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xb45708?, 0xc000111380?}, 0xc000111380?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:250 +0x31
net/http.HandlerFunc.ServeHTTP(0x961460?, {0xb45708?, 0xc000111380?}, 0x4?)
	net/http/server.go:2166 +0x29
github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xb44b38, 0xc00023e000}, 0xc0001d0120)
	github.com/klauspost/compress@v1.17.8/gzhttp/compress.go:501 +0x453
net/http.HandlerFunc.ServeHTTP(0x40eca5?, {0xb44b38?, 0xc00023e000?}, 0xc00023e001?)
	net/http/server.go:2166 +0x29
net/http.serverHandler.ServeHTTP({0xb439a0?}, {0xb44b38?, 0xc00023e000?}, 0x6?)
	net/http/server.go:3137 +0x8e
net/http.(*conn).serve(0xc00021c090, {0xb463c8, 0xc0000c6360})
	net/http/server.go:2039 +0x5e8
created by net/http.(*Server).Serve in goroutine 57
	net/http/server.go:3285 +0x4b4

**Comment by Haleygo** [comment]
Hello @zoutao0909 ,
could you share the value of metric `process_cpu_cores_available`? It can be see on `http://localhost:9248/metrics`.

**Comment by zoutao0909** [comment]
Thanks for the reply. The indicators I queried are as follows: process_cpu_cores_available 0.5

**Comment by Haleygo** [comment]
@zoutao0909 could you also check `vm_available_cpu_cores`?
And how do you run VictoriaLog here, using binary or container?

**Comment by zoutao0909** [comment]
The indicators I queried are as follows：vm_available_cpu_cores 1
and run VictoriaLog using container.

my container config:

```
containers:
  - args:
    - -httpListenAddr=0.0.0.0:9248
    - -retentionPeriod=30d
    - -storageDataPath=/storage
    - -memory.allowedPercent=60
    - -inmemoryDataFlushInterval=1s
    - -prevCacheRemovalPercent=1
    - -cacheExpireDuration=5m
    image: victoriametrics/victoria-logs:v0.8.0-victorialogs
    imagePullPolicy: Always
    name: victor-log
    ports:
    - containerPort: 9428
      name: http
      protocol: TCP
    resources:
      limits:
        cpu: 500m
        memory: 500Mi
      requests:
        cpu: 10m
        memory: 10Ki
    securityContext:
      capabilities:
        add:
        - SYS_PTRACE
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/localtime
      name: localtime
    - mountPath: /storage
      name: data
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-lxlhk
      readOnly: true
```

**Comment by Haleygo** [comment]
Should be fixed in https://github.com/VictoriaMetrics/VictoriaMetrics/commit/dc55146752c4c3d237ba2a3143694e0f768e13bc, and the fix is included from [v0.12.1-victorialogs](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v0.12.1-victorialogs), please try v0.12.1 or newer version.
Close as completed, feel free to reopen if there is still problem.
